dvc/utils/collections.py

```diff
@@ -61,6 +51,53 @@ def is_same_type(a, b):
         )
 
 
+def to_omegaconf(item):
```

Comment:
All these 3 functions had to be added because we try to preserve comments while modifying the params file.

I don't know if this comment-preserving logic was a user request or how relevant it actually is. 
For example,  [YAML spec](https://yaml.org/spec/1.2.2/#3233-comments) says:

```
Comments are a presentation detail and must not have any effect on the serialization tree or representation graph. In particular, comments are not associated with a particular node.
```

If we discard the idea of trying to preserve comments we could remote quite a lot of code